### ISSUE TITLE
Probabilities with loss_function=MultiClass and loss_function=Logloss with binary labels are not the same

### ISSUE THREAD

**Issue by dvpolyakov** [catboost/catboost#807]
Problem: in classification task when the labels in the target vector are in range {0,1} if one set loss_function=MultiClass and loss_function=Logloss, he will get different predicted probabilities.

Detailed explanation with code is here: https://github.com/dvpolyakov/CatBoost_issues/blob/master/Catboost%20usability%20example%20issue%202.ipynb

catboost version: 0.14.1
Operating System: Ubuntu

**Comment by annaveronika** [comment]
In your case the reason is that binary classification sets a default learning_rate which depends on your data and number of iterations. And multi-classification doesn't support this yet and has a default learning rate 0.03. So binary classification was trained with a much larger learning rate, since you specified 5 iterations.

But multi-classification and binary classification are also not equal modes in terms of implementation - in multiclass mode you have a formula value for each class, and then you calculate probabilities as a softmax, and in binary classification you have a single value. So there will be some difference in values.

I think we will run binary classification or multi classification by default in CatBoostClassifier based on count of classes in label values. But multiclass will still be available for dataset with 2 class values.
